src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/IdApplicationService.kt

```diff
@@ -21,11 +21,15 @@ class IdApplicationService(
 ) {
 
   @Transactional
-  suspend fun getIdApplicationByNomsId(nomsId: String): IdApplicationEntity {
+  suspend fun getIdApplicationByNomsId(nomsId: String): IdApplicationEntity? {
     val prisoner = prisonerRepository.findByNomsId(nomsId)
       ?: throw ResourceNotFoundException("Prisoner with id $nomsId not found in database")
-    return idApplicationRepository.findByPrisonerAndIsDeleted(prisoner)
-      ?: throw ResourceNotFoundException("No active ID application found for prisoner with id $nomsId")
+    val idApplicationEntityList = idApplicationRepository.findByPrisonerAndIsDeleted(prisoner)
+    if (idApplicationEntityList.isEmpty()) {
+      throw ResourceNotFoundException("No active ID application found for prisoner with id $nomsId")
+    } else {
+      return idApplicationEntityList[0]
```

Comment:
how come we are just returning the first item?

Reply:
We have separate endpoint  to return multiple values, this endpoint to return one. I reckon this endpoint (/{nomsId}/idapplication) need to be deprecated and FE to use the new endpoint (/{nomsId}/idapplication/all)